### DISCUSSION TITLE
Is it possible to specify the namespace prefixes and the default namespace

### DISCUSSION THREAD

**Discussion by couling** [tefra/xsdata#922] [Q&A] [answered]
I'm searching the documentation regarding namespace prefixes, but not finding anything which works.  When serialising, I'm looking for a way to set the namespace prefix for specific namespaces and set the default namespace.

For example I don't want to generate this:

```xml
<ns0:foo xmlns:ns0="http://bar.example.com" />
```

I want to be able to specify either the default namespace:
```xml
<foo xmlns="http://bar.example.com" />
```

Or force the namespace to be a prefix of my chosing (`bar` in this example)
```xml
<bar:foo xmlns:bar="http://bar.example.com" />
```

------

I realise that in terms of XML standards these are all equivalent.  However there are still broken interpreters around that choke on the "wrong" prefix.

**Comment by AlexandrosPlessias** [comment, marked as answer]
Hello @couling,

You can use below codesnap:

```
from xsdata.formats.dataclass.serializers.xml import XmlSerializer
from xsdata.formats.dataclass.serializers.config import SerializerConfig

serializer_config: SerializerConfig = SerializerConfig(pretty_print=True)
xml_serializer: XmlSerializer = XmlSerializer(config=serializer_config)
```

The foo param is the root class of your xml & ns_map can be a dict variable if you want 

For 
`<foo xmlns="http://bar.example.com" />`
try below: 
`xml_foo: str = xml_serializer.render(foo, ns_map={"": "http://bar.example.com"})`

For 
`<bar:foo xmlns:bar="http://bar.example.com" />`
 try below: 
`xml_foo: str = xml_serializer.render(foo, ns_map={"": "http://bar.example.com","bar":"http://bar.example.com"})`


Here you write the xml file:
```
with open(f"{props.OUTPUT_PATH}{export_filename}.xml", "r") as f:
     f.write(xml_foo)
```
